Mathlib/Geometry/Manifold/SmoothManifoldWithCorners.lean

```diff
@@ -1398,6 +1398,31 @@ theorem ContinuousWithinAt.extChartAt_symm_preimage_inter_range_eventuallyEq
   rw [← nhdsWithin_eq_iff_eventuallyEq]
   exact hc.nhdsWithin_extChartAt_symm_preimage_inter_range
 
+section LocallyCompact
+
```

Comment:
Let's add some docs:
```sugestion
/-- A locally compact manifold must be modelled on a locally compact space. -/
```

Can't you show this result for manifolds with boundary already? Just choose some interior point, and the same argument should work, right? (A new condition on `ModelWithCorners` ensures that interior points are dense, at least in the model space.)

Does `extChartAt_image_nhd_mem_nhds_of_boundaryless` require the interior to be open? It looks to me like it may not. (If it does, that's a fine reason to wait until a better theory of interior of a manifold is developed. I'm not convinced yet that's required, though.)

Reply:
I wrote that at 4am. 😬 Let me look into this.

Reply:
The more general fact that leads to this is: if a manifold `M` is non-empty, then its interior is non-empty, i.e. no boundary point can exist without interior points. I'm inclined to leave that and the generalisation of the current theorem as a TODO for a future PR.